recipe/conda_build_config.yaml

```diff
@@ -0,0 +1,2 @@
+MACOSX_SDK_VERSION:  # [osx and arm64]
+  - "13.0"           # [osx and arm64]
```

Comment:
We are seeing [this error on CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1197254&view=logs&j=a53b79f6-c5b6-55b8-410d-8b93340e372e&t=499d01a2-61e4-510e-3cc0-54b91fc59285&l=1211 )

```
2025-03-13 02:10:02.235 xcodebuild[11872:47196] [MT] DVTSDK: Skipped SDK /Applications/Xcode_15.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk; its version (11.0) is below required minimum (13.0) for the macosx platform.
2025-03-13 02:10:03.623 xcodebuild[11872:47196] Writing error result bundle to /var/folders/mj/dx2cc6hn6h9br7vt9hz4n06m0000gn/T/ResultBundle_2025-13-03_02-10-0003.xcresult
```

Hence this is bumped to macOS 13.0 as the minimum compatible version